src/edu/csus/ecs/pc2/core/list/RunList.java

```diff
@@ -338,17 +363,19 @@ private int lastRunNumber(int siteNumber) {
     }
 
     public int size() {
-        synchronized (runHash) {
+        synchronized (runHashLock) {
             return runHash.size();
         }
     }
 
     public Run[] getList() {
-        if (runHash.size() == 0) {
-            return new Run[0];
-        }
+        synchronized (runHashLock) {
+            if (runHash.size() == 0) {
+                return new Run[0];
+            }
 
-        return (Run[]) runHash.values().toArray(new Run[size()]);
+            return (Run[]) runHash.values().toArray(new Run[size()]);
```

Comment:
NPE/Design note - the reason for the synchronized statements is that the toArray statement and
its iterator were throwing a NPE because of a natural stress test.   One hypothosis is that
the update of the runHash was deleting (nulling out) elements returned by the iterator in toArray.